src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,55 @@
 package core.basesyntax;
 
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+
 public class WorkWithFile {
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT = "result";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String toFile = readFromFile(fromFileName);
+        String report = createReport(toFile);
+        writeToFile(toFileName, report);
+    }
+
+    private String readFromFile(String fileName) {
+        File file = new File(fileName);
+        try {
+            return Files.readAllLines(file.toPath()).toString();
+        } catch (IOException e) {
+            throw new RuntimeException("Can`t read file " + fileName, e);
+        }
+    }
+
+    private String createReport(String strings) {
+        String[] strings1 = strings.split("\\W+");
```

Comment:
```suggestion
        String[] tokens = strings.split(DELIMITER_REGEX );
```
Please pay attention to the naming, it's important. Remember, that developers spend way more time reading code than writing it